python/semantic_kernel/planners/function_calling_stepwise_planner/function_calling_stepwise_planner.py

```diff
@@ -0,0 +1,249 @@
+                args = json.loads(chat_result.tool_calls[0].function.arguments)
+                answer = args["answer"]
+                return FunctionCallingStepwisePlannerResult(
+                    final_answer=answer,
+                    chat_history=chat_history_for_steps,
+                    iterations=i + 1,
+                )
+
+            chat_history_for_steps = await self._process_tool_calls(chat_result, cloned_kernel, chat_history_for_steps)
+
+        # We're done, but the model hasn't returned a final answer.
+        return FunctionCallingStepwisePlannerResult(
+            final_answer="",
+            chat_history=chat_history_for_steps,
+            iterations=i + 1,
+        )
+
+    async def _process_tool_calls(
+        self,
+        chat_result: OpenAIChatMessageContent,
+        kernel: Kernel,
+        chat_history: ChatHistory,
+    ) -> ChatHistory:
+        """Process the tool calls in the chat result and return the chat history for the steps"""
+        logger.info(f"processing {len(chat_result.tool_calls)} tool calls")
+        for tool_call in chat_result.tool_calls:
+            func = kernel.func(**tool_call.function.split_name_dict())
+            if func is None:
+                chat_history.add_user_message(
+                    f"Function {tool_call.function.name} not found in the kernel. Try something else!"
+                )
+                continue
+            arguments = tool_call.function.to_kernel_arguments()
+            logger.info(f"Calling {tool_call.function.name} function with args: {arguments}")
+            func_result = await kernel.invoke(func, arguments)
+            chat_history.add_tool_message(
+                str(func_result.value),
+                metadata={"tool_call_id": tool_call.id, "function_name": tool_call.function.name},
+            )
+        return chat_history
+
+    async def _get_completion_with_functions(
+        self,
+        chat_history: ChatHistory,
+        kernel: Kernel,
+        chat_completion: ChatCompletionClientBase,
+        execution_settings: OpenAIChatPromptExecutionSettings,
+    ) -> OpenAIChatMessageContent:
+        """Get the completions with functions"""
+        return await chat_completion.complete_chat(
+            chat_history=chat_history,
+            settings=execution_settings,
+            kernel=kernel,
+        )
+
+    async def _build_chat_history_for_step(
+        self,
+        goal: str,
+        initial_plan: str,
+        kernel: Kernel,
+    ) -> ChatHistory:
+        """Build the chat history for the stepwise planner"""
+        chat_history = ChatHistory()
+        arguments = KernelArguments(
+            goal=goal,
+            initial_plan=initial_plan,
+        )
+        prompt_template_config = PromptTemplateConfig(
+            template=self.step_prompt,
+        )
+        kernel_prompt_template = KernelPromptTemplate(prompt_template_config=prompt_template_config)
```

Comment:
here its easier to just put the prompt into the KernelPromptTemplate!

Reply:
Not sure I follow? The KernelPromptTemplate need a prompt template config...?

Reply:
No it can have just a prompt, it then takes care of the rest!

Reply:
ah, we should make that more apparent in the docstring, because even I didn't realize that :)